src/renderer/api/kube-api.ts

```diff
@@ -86,15 +93,36 @@ export class KubeApi<T extends KubeObject = any> {
     this.apiPrefix = apiPrefix;
     this.apiGroup = apiGroup;
     this.apiVersion = apiVersion;
-    this.apiVersionWithGroup = apiVersionWithGroup;
     this.apiResource = resource;
     this.request = request;
     this.objectConstructor = objectConstructor;
 
+    this.checkPreferredVersion();
     this.parseResponse = this.parseResponse.bind(this);
     apiManager.registerApi(apiBase, this);
   }
 
+  get apiVersionWithGroup() {
+    return [this.apiGroup, this.apiVersionPreferred ?? this.apiVersion]
+      .filter(Boolean)
+      .join("/")
+  }
+
+  protected async checkPreferredVersion() {
+    if (!this.options.checkPreferredVersion || this.apiVersionPreferred === undefined) {
+      return;
```

Comment:
isn't this always returning since we don't update `apiVersionPreferred` elsewhere?

Reply:
Oh, right. Have to tweak `if` condition..